src/widgets/courses/ui/courses.tsx

```diff
@@ -13,19 +15,13 @@ export const Courses = () => {
   if (error) return <h2>{error.message}</h2>;
   if (!courses || courses.length === 0) return null;
 
-  const nearestCourse = finedNearestCourse(courses);
-  const nearestCourseStartDate = nearestCourse ? Date.parse(nearestCourse.startDate) : Date.now();
+  const sortParams = {
+    dataList: courses.filter(isCourse),
+    actualDelayInDays: COURSE_STALE_AFTER_DAYS,
+    filtered: false,
+  };
 
-  const sortedCourses = courses
-    .filter(isCourse)
-    .sort((a, b) => compareNumbers(a, b, nearestCourseStartDate))
-    .map((course) => {
-      return {
-        ...course,
-        startDate:
-          nearestCourseStartDate <= Date.parse(course.startDate) ? course.startDate : '(TBD)',
-      };
-    });
+  const sortedCourses = getActualDataList(sortParams) as Course[];
```

Comment:
I don't really like that we need to do type casting (`as Course[]`) in order to use this function, why can't we separate function for getting `Course` and another one for getting `EventCardProps`?

![image](https://github.com/rolling-scopes/site/assets/114234698/d5aa724f-95db-454e-9e63-f3df7e57fc4c)

Reply:
As @ansivgit requested, will make it in separate pr #400